### PR TITLE
Text conversion cleanup

### PR DESCRIPTION
**Pull request by jystic**
I did a clean up of the current text conversion code to make it a bit easier to understand and more consistent. I thought it would be a good idea to do this before I start writing optimised versions of these functions.

### CODE REVIEW

File: data/sea/42-text-conversion.h [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+        case  2:  value += (p[n- 2] - '0') * 10;
+        case  1:  value += (p[n- 1] - '0');
+                  value *= sign;
+                  *output_ptr = value;
+                  *pp += n;
+                  return 0;
+
+        default:  return ierror_msg_alloc ("integer too big", p, n);
+    }
+}
```
Comment (jystic) [inline data/sea/42-text-conversion.h]: This new `text_read_iint` speeds up the ASX 500MB dataset from 410MB/s -> 630MB/s on my computer.

File: data/sea/42-text-conversion.h [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+{
+    char *p            = *pp;
+    size_t buffer_size = pe - p;
+
+    size_t n = 0;
+    while (n < buffer_size) {
+        if (!text_is_digit (p[n]))
+            break;
+        n++;
+    }
```
Comment (jystic) [inline data/sea/42-text-conversion.h]: I'm hoping to replace this with a call to `_mm_cmpestr*`

**Comment by olorin** [comment]
Nice. (From a cursory glance, looks like a lot of the complexity in the `glibc` implementation is the base handling.)

**Comment by amosr** [comment]
looks good :+1:

**Comment by amosr** [comment]
somewhat unrelated, but while you're in there, perhaps make sure this time reading code won't dereference before the start of the buffer. (I guess you could also pad the start of the buffer if you wanted?)
https://github.com/ambiata/icicle/blob/topic/text-conversion-cleanup/data/sea/52-psv.h#L116

**Comment by jystic** [comment]
I wonder if it's ok because it's on the stack? so it's already padded in a sense

**Comment by amosr** [comment]
hmm... yes, I suppose so. the stack grows downwards right, so it'd probably be padded by variables in later functions? I'm not sure, I think I'd rather do it explicitly

**Comment by jystic** [comment]
So something like adding 16 bytes to the start and the end of the buffer?

**Pull request merged**